src/Microsoft.Data.Analysis/AssemblyAttributes.cs

```diff
@@ -0,0 +1,7 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.Versioning;
+
+[assembly: RequiresPreviewFeatures]
```

Comment:
Is this because you're targeting .NET 6?

The API surface of the generic math interfaces changed between 6 and 7 (such as moving all interfaces into `namespace System.Numerics`) and it would likely be better to keep this targeting 7/8

Reply:
Yeah - figured targeting the LTS made sense. Thoughts?

Reply:
I think if we're making this change, we'd want to target .NET 8 (LTS) and plan to ship it around or after November

Reply:
The approach we have recently been taking with NuGet packages here and in dotnet/runtime is to retain support for in-support frameworks.  This is better than the previous approach which was never drop support for anything due to compatibility.  .NET 6 remains [in support until November of 2024](https://dotnet.microsoft.com/en-us/platform/support/policy/dotnet-core#lifecycle) so our current plan would be to keep supporting it until then.

ML.NET's servicing story so far has been "update to latest" - we only maintain one active release branch.  I could imagine a change where we start dropping in-support frameworks from our nuget package, but in conjunction with such a change we'd need to maintain more servicing branches, and as a result servicing changes would become more expensive since we'd need to port them between many codebases.

Reply:
I think this would be a special consideration for .NET 6 vs .NET 8 and won't be "normal" as we don't often get features as big as generic math.

So we'd need to maintain a servicing branch for 6, but then wouldn't need to the same for 8 vs 10 without some other feature that is as meaningfully impactful

Reply:
In this case what's the benefit from not supporting 6.0?  Is it just removing that attribute or is there more to it?

Reply:
There were concrete breaking changes to the API surface of the generic math feature between .NET 6 and .NET 7

We'd be maintaining two copies of the generic math code, one in preview and one "feature complete" as well as dealing with any subtle bugs or missing support in the broader product from it being an early preview.

Reply:
This PR also removes support for netstandard2.0 and full .Net frameworks (like 4.8, which is as far as I undertans also LTS).  The company I worked before, used DataFrame on .Net 4.6.2 and they didn't have plan to migrate to .net core in the nearest future. 
Could you please make at least one more release of Microsoft.Data.Analysis nuget targeting netstandart2.0 for such clients before moving to ,Net 8 (6) only?

Reply:
@asmirnov82 we're still figuring out the timelines for this change but it will be after the next release.

Thanks for working with us!